sklearn/base.py

```diff
@@ -563,3 +562,37 @@ def is_regressor(estimator):
         True if estimator is a regressor and False otherwise.
     """
     return getattr(estimator, "_estimator_type", None) == "regressor"
+
+
+def is_pairwise(estimator):
+    """Returns True if estimator has a _pairwise attribute set to True.
+
+    Parameters
+    ----------
+    estimator : object
+        Estimator object to test.
+
+    Returns
+    -------
+    out : bool
+        True if _pairwise is set to True and False otherwise.
+    """
+    return bool(getattr(estimator, "_pairwise", False))
+
+
+def is_pairwise_metric(estimator):
+    """Returns True if estimator has a _pairwise attribute set to True.
+
+    Parameters
+    ----------
+    estimator : object
+        Estimator object to test.
+
+    Returns
+    -------
+    out : bool
+        True if _pairwise is set to True and False otherwise.
+    """
+    metric = getattr(estimator,  "metric", None)
+
+    return metric == 'precomputed' and is_pairwise(estimator)
```

Comment:
I'm not certain we need `metric == 'precomputed'` here. What made you think it necessary/appropriate?

Reply:
Was trying to find a proxy for whether or not an estimator requires a distance or affinity metric. Do estimators have a property that does this explicitly?

Reply:
But if is_pairwise() is True, then presence of metric as a param might be enough?